File: guides/java/kora-java-guide-database-jdbc-advanced-app/src/main/java/io/koraframework/guide/databasejdbc/advanced/task/dto/TaskRequest.java
package io.koraframework.guide.databasejdbc.advanced.task.dto;

import org.jspecify.annotations.Nullable;
import io.koraframework.json.common.annotation.Json;

import java.util.List;

@Json
public record TaskRequest(List<TaskCreate> tasks) {

    @Json
    public record TaskCreate(String title,
                             @Nullable String description,
                             @Nullable Long userAssigneeId) { }
}

File: guides/kotlin/kora-kotlin-guide-database-jdbc-advanced-app/src/main/kotlin/io/koraframework/guide/databasejdbc/advanced/task/dto/TaskRequest.kt
package io.koraframework.guide.databasejdbc.advanced.task.dto

import io.koraframework.json.common.annotation.Json

@Json
data class TaskRequest(
    val tasks: List<TaskCreate>
) {
    @Json
    data class TaskCreate(
        val title: String,
        val description: String?,
        val userAssigneeId: Long?
    )
}
